src/client/Client.cc

```diff
@@ -9243,14 +9243,14 @@ void Client::seekdir(dir_result_t *dirp, loff_t offset)
 //};
 void Client::fill_dirent(struct dirent *de, const char *name, int type, uint64_t ino, loff_t next_off)
 {
-  strncpy(de->d_name, name, 255);
+  char *last = stpncpy(de->d_name, name, 255);
   de->d_name[255] = '\0';
 #if !defined(__CYGWIN__) && !(defined(_WIN32))
   de->d_ino = ino;
 #if !defined(__APPLE__) && !defined(__FreeBSD__)
   de->d_off = next_off;
 #endif
-  de->d_reclen = 1;
+  de->d_reclen = (uintptr_t)last - (uintptr_t)de + 1;
```

Comment:
Good catch. Just FYI - I had discussed this with @xhernandez over chat and the issue cropped up when developing proxy b/w samba and libcephfs where d_reclen was relied on.